src/lualib/ArrayReduce.ts

```diff
@@ -0,0 +1,28 @@
+// https://www.ecma-international.org/ecma-262/9.0/index.html#sec-array.prototype.reduce
+function __TS__ArrayReduce<T>(
+    this: void,
+    arr: T[],
+    callbackFn: (accumulator: T, currentValue: T, index: number, array: T[]) => T,
+    initial?: T
+): T {
+    const len = arr.length;
+
+    if (len === 0 && initial === undefined) {
+        // tslint:disable-next-line: no-string-throw
+        throw "Cannot reduce empty list without initial value.";
```

Comment:
```suggestion
        throw "Reduce of empty array with no initial value";
```
It's not a part of standard, but still would be nice to follow a common practice: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Reduce_of_empty_array_with_no_initial_value